src/cbc_sdk/platform/jobs.py

```diff
@@ -84,6 +85,45 @@ def get_progress(self):
         self._info['progress'] = resp
         return resp['num_total'], resp['num_completed'], resp.get('message', None)
 
+    def _await_completion(self):
+        """
+        Waits for this job to complete by examining the progress data.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Returns:
+            Job: This object.
+        """
+        progress_data = (1, 0, '')
+        sleep_time = 0.0
+        while progress_data[1] < progress_data[0]:
+            if sleep_time > 0.0:
+                time.sleep(sleep_time)
+            try:
+                progress_data = self.get_progress()
+            except ServerError:
+                progress_data = (1, 0, '')
+            if sleep_time > 0.0:
```

Comment:
Can you add a comment here that the sleep time progressively doubles until it hits 1s where it maintains a 1s sleep time. This will provide clear and immediate understanding of the logic in the future

Reply:
"The sleep time starts at 0.1 seconds and doubles each time until it hits 1.6 seconds, providing a simple exponential backoff between progress calls."

Reply:
Hmm. I see your point about how 0.8 will double and be greater than 1.0 though as a user I would ask why the arbitrary 1.6s? Why not keep it simple and use a flat 0.5s or 1s delay? If possible why not make this configurable? My intention for requesting a comment here is to provide context so in the future we understand any reasoning for why we added this mechanism for delay. Should this be a helper function to provide consistent exponential backoff?

Reply:
We're leaving the issue of exponential backoff for another time and changing it to a flat 0.5s delay for now.